lib/matplotlib/colorizer.py

```diff
@@ -0,0 +1,682 @@
+        return rgba
+
+    @staticmethod
+    def _pass_image_data(x, alpha=None, bytes=False, norm=True):
+        """
+        Helper function to pass ndarray of shape (...,3) or (..., 4)
+        through `to_rgba()`, see `to_rgba()` for docstring.
+        """
+        if x.shape[2] == 3:
+            if alpha is None:
+                alpha = 1
+            if x.dtype == np.uint8:
+                alpha = np.uint8(alpha * 255)
+            m, n = x.shape[:2]
+            xx = np.empty(shape=(m, n, 4), dtype=x.dtype)
+            xx[:, :, :3] = x
+            xx[:, :, 3] = alpha
+        elif x.shape[2] == 4:
+            xx = x
+        else:
+            raise ValueError("Third dimension must be 3 or 4")
+        if xx.dtype.kind == 'f':
+            # If any of R, G, B, or A is nan, set to 0
+            if np.any(nans := np.isnan(x)):
+                if x.shape[2] == 4:
+                    xx = xx.copy()
+                xx[np.any(nans, axis=2), :] = 0
+
+            if norm and (xx.max() > 1 or xx.min() < 0):
+                raise ValueError("Floating point image RGB values "
+                                 "must be in the 0..1 range.")
+            if bytes:
+                xx = (xx * 255).astype(np.uint8)
+        elif xx.dtype == np.uint8:
+            if not bytes:
+                xx = xx.astype(np.float32) / 255
+        else:
+            raise ValueError("Image RGB array must be uint8 or "
+                             "floating point; found %s" % xx.dtype)
+        # Account for any masked entries in the original array
+        # If any of R, G, B, or A are masked for an entry, we set alpha to 0
+        if np.ma.is_masked(x):
+            xx[np.any(np.ma.getmaskarray(x), axis=2), 3] = 0
+        return xx
+
+    def autoscale(self, A):
+        """
+        Autoscale the scalar limits on the norm instance using the
+        current array
+        """
+        if A is None:
+            raise TypeError('You must first set_array for mappable')
+        # If the norm's limits are updated self.changed() will be called
+        # through the callbacks attached to the norm
+        self.norm.autoscale(A)
+
+    def autoscale_None(self, A):
+        """
+        Autoscale the scalar limits on the norm instance using the
+        current array, changing only limits that are None
+        """
+        if A is None:
+            raise TypeError('You must first set_array for mappable')
+        # If the norm's limits are updated self.changed() will be called
+        # through the callbacks attached to the norm
+        self.norm.autoscale_None(A)
+
+    def _set_cmap(self, cmap):
+        """
+        Set the colormap for luminance data.
+
+        Parameters
+        ----------
+        cmap : `.Colormap` or str or None
+        """
+        # bury import to avoid circular imports
+        from matplotlib import cm
+        in_init = self._cmap is None
+        cmap = cm._ensure_cmap(cmap)
+        self._cmap = cmap
```

Comment:
Should fit together like it used to.
```suggestion
        self._cmap = cm._ensure_cmap(cmap)
```